provider/waiter.go

```diff
@@ -102,104 +103,88 @@ type FieldWaiter struct {
 	resourceName  string
 	resourceType  tftypes.Type
 	fieldMatchers []FieldMatcher
+	logger        hclog.Logger
 }
 
 // Wait blocks until all of the FieldMatchers configured evaluate to true
 func (w *FieldWaiter) Wait(ctx context.Context) error {
-	return wait(ctx, w.resource, w.resourceName, w.resourceType, func(obj tftypes.Value) (bool, error) {
-		for _, m := range w.fieldMatchers {
-			vi, rp, err := tftypes.WalkAttributePath(obj, m.path)
-			if err != nil {
-				return false, err
-			}
-			if len(rp.Steps) > 0 {
-				return false, fmt.Errorf("attribute not present at path '%s'", m.path.String())
-			}
-
-			var s string
-			v := vi.(tftypes.Value)
-			switch {
-			case v.Type().Is(tftypes.String):
-				v.As(&s)
-			case v.Type().Is(tftypes.Bool):
-				var vb bool
-				v.As(&vb)
-				s = fmt.Sprintf("%t", vb)
-			case v.Type().Is(tftypes.Number):
-				var f big.Float
-				v.As(&f)
-				if f.IsInt() {
-					i, _ := f.Int64()
-					s = fmt.Sprintf("%d", i)
-				} else {
-					i, _ := f.Float64()
-					s = fmt.Sprintf("%f", i)
-				}
-			default:
-				return true, fmt.Errorf("wait_for: cannot match on type %q", v.Type().String())
-			}
-
-			log.Printf("matching %#v %q", m.valueMatcher, s)
-
-			if !m.valueMatcher.Match([]byte(s)) {
-				return false, nil
-			}
-		}
-
-		return true, nil
-	})
-}
-
-// NoopWaiter is a placeholder for when there is nothing to wait on
-type NoopWaiter struct{}
-
-// Wait returns immediately
-func (w *NoopWaiter) Wait(_ context.Context) error {
-	return nil
-}
-
-func wait(ctx context.Context, resource dynamic.ResourceInterface, resourceName string, rtype tftypes.Type, condition func(tftypes.Value) (bool, error)) error {
-	w, err := resource.Watch(ctx, v1.ListOptions{
-		FieldSelector: fields.OneTermEqualSelector("metadata.name", resourceName).String(),
-		Watch:         true,
-	})
-	if err != nil {
-		return err
-	}
-
-	log.Printf("[ApplyResourceChange][Wait] Waiting until ready...\n")
-	for e := range w.ResultChan() {
-		if e.Type == watch.Deleted {
-			return fmt.Errorf("resource was deleted")
-		}
-
-		if e.Type == watch.Error {
-			log.Printf("Error when watching: %#v", e.Object)
-			return fmt.Errorf("watch error")
-		}
-
+	w.logger.Info("[ApplyResourceChange][Wait] Waiting until ready...\n")
+	for {
 		// NOTE The typed API resource is actually returned in the
```

Comment:
We can delete this comment since we're not using the event stream anymore.

Reply:
But I feel like it still signals the activity of running the matchers. Wasn't that it's intended meaning?

Reply:
Not sure I understand you – the reason I left this comment was to explain why I was doing another `Get()` even though the contents of the resource would be inside the event object. It doesn't make sense anymore because we've removed the event stream code and are just retrying Get until everything matches.